### ISSUE TITLE
[react-jss] Merging classes instead of an override and optionally override

### ISSUE THREAD

**Issue by kof** [cssinjs/jss#933]
People ask for a way to merge the classes when using the HOC interface instead of overriding them. If thats what everyone wants to do, we could implement it by default.

https://stackoverflow.com/questions/53674340/merge-classes-with-jss-instead-of-overriding/53674874

## Optional override

We could additionally support an override like this: 

```js
classes = {
   a: 'class-a', // extends
   b: ['class-b' , true], // overrides
   c: ['class-c', 'class-d'  , true] // overrides and bonus - one can use array of classes
}
```

**Comment by pswg** [comment]
Just brainstorming a little. If `classes` could accept a function, you could use a syntax like this:

```
import _mapKeys from 'lodash/mapKeys';

const mergeClasses =
  (extends, overrides) =>
    (baseClasses) =>
      _mapKeys((cls, key) =>
        key in overrides ? overrides[key] : key in extends ? cls + ' ' + extends[key] : cls);

...

classes = {
  mergeClasses({
    a: 'class-a'
  }, {
    b: 'class-b',
    c: 'class-c class-d'
  })
}
```

Not sure if this makes anything easier or much more difficult, though.

**Comment by kof** [comment]
Yes, but would you want to do it as a consumer every time?

**Comment by pswg** [comment]
Probably not, just proposing an alternate syntax. If it *were* implemented this way, I would propose that a comparable `mergeClasses` function be added to the `react-jss` library so the only extra burden to users would be an import then calling it in `classes`.

**Comment by oliviertassinari** [comment]
This sounds like a good tradeoff, for reference, the `mergeClasses` method on Material-UI side:
https://github.com/mui-org/material-ui/blob/5be50c86e9ba5ca08659e4bef14aea2b8bc2d902/packages/material-ui-styles/src/mergeClasses.js

**Comment by kof** [comment]
The question is what is the mainstream behaviour. If everyone wants mergeClasses behaviour often, asking to provide a function and then import mergeClasses and merge it in consumer component is a lot of work.

**Comment by HenriBeck** [comment]
If we want to support both merging and overriding, I think the syntax from @kof is the cleanest and easiest.

But I'm not quite sure about allowing multiple classes inside the array.

**Comment by HenriBeck** [comment]
Implemented in #946
